ipa-core/src/protocol/ipa_prf/aggregation/bucket.rs

```diff
@@ -116,26 +115,25 @@ where
         }
 
         let depth_c = ctx.narrow(&BinaryTreeDepthStep::from(tree_depth));
-        let mut futures = Vec::with_capacity(breakdown_count / step);
-
-        for (i, tree_index) in (0..breakdown_count).step_by(step).enumerate() {
-            let bucket_c = depth_c.narrow(&BucketStep::from(i));
-
-            let index_contribution = row_contribution[tree_index].iter();
-
-            if robust || tree_index + span < breakdown_count {
-                futures.push(async move {
-                    let bit_futures = index_contribution
-                        .zip(repeat((bit_of_bdkey, bucket_c.clone())))
-                        .enumerate()
-                        .map(|(i, (a, (b, bucket_c)))| {
-                            a.multiply(b, bucket_c.narrow(&BitOpStep::Bit(i)), record_id)
-                        });
-                    BitDecomposed::try_from(bucket_c.parallel_join(bit_futures).await?)
-                });
-            }
-        }
-        let contributions = ctx.parallel_join(futures).await?;
+
+        let contributions = ctx
+            .parallel_join((0..breakdown_count).step_by(step).enumerate().filter_map(
+                |(i, tree_index)| {
+                    let bucket_c = depth_c.narrow(&BucketStep::from(i));
+
+                    let index_contribution = &row_contribution[tree_index];
+
+                    (robust || tree_index + span < breakdown_count).then(|| {
+                        bool_and(
+                            bucket_c,
+                            record_id,
+                            index_contribution,
+                            repeat_n(bit_of_bdkey, index_contribution.len()),
```

Comment:
OK, this surprised me. Based on the type signature of `bool_and`, I was expecting this argument to be an iterator of `AdditiveShare<Boolean, N>`, but as far as I can tell, `bit_of_bdkey` is just a single bit. We get it from `bd_key.iter()`, so it seems like that would just be one bit. Can you explain to me what I'm missing?

Reply:
`bit_of_bdkey` is an `AdditiveShare<Boolean, N>`. Like `BitOpStep`, this is describing how the protocol would operate using language reflecting a single record.

Maybe a way of making it clearer without excessive verbosity would be a `v_` prefix or suffix on variables that are carrying vectorized data?

Reply:
Hmm... Perhaps this approach of using language to describe it as if it was not vectorized, when it actually is, is worth reconsidering. Maybe we just lean into vectorisation and re-name methods to make it obvious when vectorisation is at play. I encountered another one of these situations today and it took me a little while to realise how it worked.

I think I like this naming convention idea of `v_` prefixes.